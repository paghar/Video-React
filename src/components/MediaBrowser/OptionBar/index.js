//@flow
import React from 'react';
import Typography from 'components/shared/Typography';
import ThumbnailView from 'components/shared/Button';
import { SortBtn, OptionBarWrapper } from './index.style';
import { Generals } from 'theme/colors';
import { Display2 } from 'theme/fonts';

type Props = {
  fileNumber: string,
  fileSize: string,
  clicked?: () => void,
};

function OptionBar({ clicked, fileNumber, fileSize }: Props) {
  return (
    <OptionBarWrapper>
      <Typography font={Display2} color={Generals.vi}>
        {`${fileNumber} Files, ${fileSize}`}
      </Typography>

      <SortBtn
        rtl
        backgroundColor={Generals.ii}
        hoverColor={Generals.viii}
        icon="61703"
        iconColor={Generals.vii}
        spacing="4.8px"
        iconSize="16px"
        width="65px"
        height="29px"
      >
        <Typography font={Display2} color={Generals.vii}>
          Name
        </Typography>
      </SortBtn>

      <ThumbnailView
        clicked={clicked}
        hoverColor={Generals.viii}
        backgroundColor={Generals.ii}
        icon="61497"
        iconSize="16px"
        iconColor={Generals.vii}
      />
    </OptionBarWrapper>
  );
}

export default OptionBar;
