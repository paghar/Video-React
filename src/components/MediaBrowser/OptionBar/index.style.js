//@flow
import styled from 'styled-components';
import Button from 'components/shared/Button';

export const OptionBarWrapper = styled.div`
  display: flex;
  align-items: center;
`;
export const SortBtn = styled(Button)`
  margin: 0px 9px 0px 12px;
`;
