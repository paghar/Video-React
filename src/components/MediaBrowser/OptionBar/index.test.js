//@flow
import React from 'react';
import renderer from 'react-test-renderer';
import OptionBar from '.';

describe('<OptionBar/>', () => {
  it('OptionBar with filenumber equal 2 digits ', () => {
    const tree = renderer
      .create(<OptionBar fileNumber="25" fileSize="800MB"></OptionBar>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('OptionBar with filenumber equal 7 digits ', () => {
    const tree = renderer
      .create(<OptionBar fileNumber="1000000" fileSize="800MB"></OptionBar>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
