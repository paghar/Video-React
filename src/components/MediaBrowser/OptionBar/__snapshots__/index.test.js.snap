// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionBar/> OptionBar with filenumber equal 2 digits  1`] = `
.c1 {
  pointer-events: auto;
}

.c4 {
  font-family: 'FontAwesome' !important;
  font-size: 16px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5 {
  font-size: 16px;
  display: block;
  padding-right: 0px;
  padding-left: 4.8px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c7 {
  font-size: 16px;
  display: block;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  direction: rtl;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  width: 65px;
  height: 29px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin: 0px 9px 0px 12px;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    25 Files, 800MB
  </span>
  <button
    className="c2 c3"
    height="29px"
    width="65px"
  >
    <span
      className="c4 c5"
      fontSize="16px"
    >
      
    </span>
    <span
      className="c1"
    >
      Name
    </span>
  </button>
  <button
    className="c6"
  >
    <span
      className="c4 c7"
      fontSize="16px"
    >
      
    </span>
  </button>
</div>
`;

exports[`<OptionBar/> OptionBar with filenumber equal 7 digits  1`] = `
.c1 {
  pointer-events: auto;
}

.c4 {
  font-family: 'FontAwesome' !important;
  font-size: 16px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5 {
  font-size: 16px;
  display: block;
  padding-right: 0px;
  padding-left: 4.8px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c7 {
  font-size: 16px;
  display: block;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  direction: rtl;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  width: 65px;
  height: 29px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c6 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin: 0px 9px 0px 12px;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    1000000 Files, 800MB
  </span>
  <button
    className="c2 c3"
    height="29px"
    width="65px"
  >
    <span
      className="c4 c5"
      fontSize="16px"
    >
      
    </span>
    <span
      className="c1"
    >
      Name
    </span>
  </button>
  <button
    className="c6"
  >
    <span
      className="c4 c7"
      fontSize="16px"
    >
      
    </span>
  </button>
</div>
`;
